src/compiler/core.ts

```diff
@@ -1399,6 +1399,32 @@ namespace ts {
     /** Returns lower case string */
     export function toLowerCase(x: string) { return x.toLowerCase(); }
 
+    const fileNameLowerCaseRegExp = /[^İi̇ıßa-z0-9\\/:\-_\. ]+/g;
+    export function toFileNameLowerCase(x: string) {

```

Comment:
Can we document why we're unconditionally handling this ourselves and not deferring to `string.toLocaleLowercase`, as we do in `compareStringsCaseSensitiveUI`, where possible (namely what cases that fails to handle)?

Reply:
https://github.com/microsoft/TypeScript/pull/36106/files#diff-66cec844e251a1918deb897eabca206bR1404 is this not good enough ?

Reply:
It explains what's being done, but not why using locale lower case is insufficient.